clients/mist_client.go

```diff
@@ -427,6 +431,9 @@ func (mc *MistClient) GetStreamInfo(streamName string) (MistStreamInfo, error) {
 }
 
 func (mc *MistClient) GetState() (MistState, error) {
+	mc.stateCacheMu.Lock()
```

Comment:
Shouldn't we just lock inserting to the cache? Then you can actually use this `RWMutex`. Otherwise it needs to wait for the whole Mist communication.

Reply:
That's the whole issue this PR is concerning, because the library uses an `RWMutex` we still see bursts of parallel requests to mist. If you think it safer I can wait for [my metric addition](https://github.com/livepeer/catalyst-api/commit/e84a3c84a6864c626949c94dbc05e9e740355ddd) to go to prod and definitively check the number of requests we're seeing before proceeding with this PR.

Reply:
Yeah, if there is no rush, then I think it's safer. 

TBH I think this _should_ be ok, but locking for longer may have some impact.